source/SpinalHDL/Simulation/simulator_specifics.rst

```diff
@@ -59,6 +59,26 @@ How SpinalHDL simulates the hardware with Synopsys VCS backend
 
 Before using VCS as the simulation backend, make sure that you have checked your system environment as :ref:`VCS environment<vcs_env>`.
 
+How SpinalHDL simulates the hardware with Xilinx XSim backend
+==============================================================
+
+1. SpinalHDL generates a Verilog/VHDL (depended on your choice) hardware model of the DUT.
+2. The HDL model is loaded in the simulator.
+3. The communication between the simulation and the JVM is established through shared memory. The commands are issued to the simulator using XSI.
+
+**Advantages:**
+
+* Support Xilinx built-in primitives and cores.
+
+**Limitations:**
+
+* Xilinx XSim is a **commercial** tool installed with Vivado. It is close source and not free. You have to own the licenses to **legally** use it.
+* Vivado versions prior to 2019.1 do not work properly.
+
+Before using XSim as the simulation backend, make sure that you have done following steps.
+1. Define VIVADO_HOME environment variable to specify where your vivado located. ex `export VIVADO_HOME=/d/Xilinx/Vivado/2022.1` (under MSYS2).
+2. Make sure two vivado path is inside the PATH. For Windows MSYS2 user, run shell command like `export PATH=$PATH:$VIVADO_HOME/bin:$VIVADO_HOME/lib/win64.o`. For Linux user just source the Vivado's settings64.sh file located at `VIVADO_HOME`.
+
```

Comment:
This is more like, what there reader is looking for :)

It is possible to provide a code snippet of a minimal SpinalSim testbench which can be used as an installation and setup check ?  It helps the reader to know if they run this program and they have everything setup, it should work.  Then in the future with feedback we can address any common error messages seen in the setup and how to fix.  Getting the environment working is probably why the reader is looking at the documentation here.

Reply:
I don't think this page is a good place to write this details. May be [this location](https://spinalhdl.github.io/SpinalDoc-RTD/master/SpinalHDL/Simulation/bootstraps.html#configuration) is better.

Reply:
It is the specific detail needed to make XSim work that would be captured here.
I don't think that is best put in the generic simulation information, but maybe a separate page per simulator is a better format.
Readers using XSim want to copy-and-paste something and know it should work as-is.  A kind of installation and setup check, something they can then communicate about if there is an error/issue they can not resolve.

Reply:
So that would be more work on this. I just keep the consistency of other parts.

Reply:
> It is the specific detail needed to make XSim work that would be captured here. I don't think that is best put in the generic simulation information, but maybe a separate page per simulator is a better format. Readers using XSim want to copy-and-paste something and know it should work as-is. A kind of installation and setup check, something they can then communicate about if there is an error/issue they can not resolve.

You're right, a simulation tutorial for each simulation engine would be a good choice.